tests/Api.IntegrationTests/Middleware/AuthorizationIntegrationTests.cs

```diff
@@ -0,0 +1,84 @@
+using System.Net;
+using System.Net.Http.Json;
+using Passwordless.Api.IntegrationTests.Helpers;
+using Passwordless.Api.IntegrationTests.Helpers.App;
+using Passwordless.Common.Models.Apps;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Passwordless.Api.IntegrationTests.Middleware;
+
+public class AuthorizationIntegrationTests : IClassFixture<PasswordlessApiFactory>
+{
+    private readonly HttpClient _client;
+
+    public AuthorizationIntegrationTests(ITestOutputHelper testOutput, PasswordlessApiFactory apiFactory)
+    {
+        apiFactory.TestOutput = testOutput;
+        _client = apiFactory.CreateClient();
+    }
+
+    [Fact]
+    public async Task ExistentEndpointProtectedByApiKey_Returns_Unauthorized_WhenBadApiKeyIsProvided()
```

Comment:
To stay consistent with our naming schema, maybe something like this (same for others):

```suggestion
    public async Task I_cannot_access_an_existing_endpoint_with_an_invalid_API_key()
```

Reply:
Seems less clear to me.